examples/simple-inference/simple-inference.h

```diff
@@ -0,0 +1,58 @@
+#pragma once
+
+// Defines sigaction on msys:
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include "common.h"
+
+#include "console.h"
+#include "llama.h"
+#include "build-info.h"
+#include "grammar-parser.h"
+
+#include <algorithm>
+#include <atomic>
+#include <limits>
+#include <cassert>
+#include <cinttypes>
+#include <cmath>
+#include <cstdio>
+#include <cstring>
+#include <ctime>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#if defined (__unix__) || (defined (__APPLE__) && defined (__MACH__))
+#include <signal.h>
+#include <unistd.h>
+#elif defined (_WIN32)
+#define WIN32_LEAN_AND_MEAN
+#ifndef NOMINMAX
+#define NOMINMAX
+#endif
+#include <windows.h>
+#include <signal.h>
+#endif
+
+#if defined(_MSC_VER)
+#pragma warning(disable: 4244 4267) // possible loss of data
+#endif
+
+#if defined (__unix__) || (defined (__APPLE__) && defined (__MACH__)) || defined (_WIN32)
+void sigint_handler(int signo);
```

Comment:
Why is this function declared in a header? I think it should be 'static' instead.

Reply:
The answer is because I'm not good at C(++). :) I just took the path of least resistance to stop the warnings. I'll probably just revert that and make all those functions `static`.

`-Wmissing-declarations` probably shouldn't be in `CFLAGS` for the examples/tests. There's no point in forward declaring stuff there and making it `static` seems kind of pointless too since nothing is going to link with them.

I'm not really worrying too much about this kind of cosmetic stuff for the seqrep pull right now because I'm pretty sure I'm going to cut a bunch of stuff and do major refactoring before I'd want to try to merge it. What I really need is feedback about the functionality so I can have a better idea of what actually is useful.

Reply:
@ggerganov was the one who suggested I put 'static' on every function besides 'main' in the examples and tests. I originally used anonymous namespaces, which is basically two lines per .cpp file. https://github.com/ggerganov/llama.cpp/pull/3184#discussion_r1326924441